AEPMedia/Sources/MediaPublicTracker.swift

```diff
@@ -160,19 +160,22 @@ class MediaPublicTracker: MediaTracker {
 
             let currentTs = self.getCurrentTimeStamp()
             if (currentTs - self.lastEventTs) > self.EVENT_TIMEOUT_MS {
-                // We have not got any public api call for 500 ms.
+                // We have not got any public api call for 1 second.
                 // We manually send an event to keep our internal processsing alive (idle tracking / ping processing).
                 self.trackInternal(eventName: MediaConstants.EventName.PLAYHEAD_UPDATE, params: self.lastPlayheadParams, internalEvent: true)
             }
         }
     }
 
     private func startTimer() {
-        if timer == nil {
-            timer = Timer.scheduledTimer(withTimeInterval: TICK_INTERVAL, repeats: true, block: { (_) in
-                self.tick()
-            })
-            timer?.fire()
+        guard self.timer == nil else {
+            return
+        }
+        
+        DispatchQueue.main.async {
```

Comment:
Is it necessary to add the scheduledTimer to the dispatch queue?  The `tick()` task already adds the trackInternal call to the dispatch queue.

Reply:
Yes, using it to schedule the timer on the main thread. Doesn't work without it.

Reply:
@kevinlind Updated the logic to use DispatchSourceTimer which can be added to the dispatchQueue that we have.